src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -0,0 +1,762 @@
+        path = self._tab_data_model.main.project_path.value
+        for fib_coord in self.correlation_target.fib_fiducials:
+            fib_coord = self.correlation_target.fib_stream.getPixelCoordinates(fib_coord.coordinates.value[0:2], check_bbox=False)
+            fib_coords.append(fib_coord)
+        fib_coords = numpy.array(fib_coords, dtype=numpy.float32)
+        for fm_coord in self.correlation_target.fm_fiducials:
+            fm_coord_2d = self.correlation_target.fm_streams[0].getPixelCoordinates(fm_coord.coordinates.value[0:2], check_bbox=False)
+            fm_coords.append([fm_coord_2d[0], fm_coord_2d[1], fm_coord.coordinates.value[2]])
+        fm_coords = numpy.array(fm_coords, dtype=numpy.float32)
+        poi_coord = self.correlation_target.fm_pois[0]
+        poi_coord_2d = self.correlation_target.fm_streams[0].getPixelCoordinates(poi_coord.coordinates.value[0:2], check_bbox=False)
+        poi_coords.append([poi_coord_2d[0], poi_coord_2d[1], poi_coord.coordinates.value[2]])
+        poi_coords = numpy.array(poi_coords, dtype=numpy.float32)
+        # Run the correlation
+        self.correlation_target.correlation_result = run_tdct_correlation(fib_coords=fib_coords, fm_coords=fm_coords,
+                                                                          poi_coords=poi_coords,
+                                                                          fib_image=fib_da, fm_image=fm_image,
+                                                                          path=path)
+        # Update the output pameters of the correlation
+        self._tab_data_model.projected_points = []
+        self.correlation_target.fib_projected_fiducials = []
+        points = self.correlation_target.correlation_result['output']['error']['reprojected_3d']
+        for n, i in enumerate(points[0]):
+            p_pos = self.correlation_target.fib_stream.getPhysicalCoordinates((points[0][n], points[1][n]))
+            target = Target(x=p_pos[0], y=p_pos[1], z=0, name="PP" + str(n + 1), type="ProjectedPoints", index=n + 1,
+                            fm_focus_position=0)
+            self._tab_data_model.projected_points.append(target)
+            self.correlation_target.fib_projected_fiducials.append(target)
+
+        # Convert from pixel to physical coordinates
+        projected_poi = (self.correlation_target.fib_stream.getPhysicalCoordinates
+                         (self.correlation_target.correlation_result["output"]["poi"][0]["image_px"]))
+        projected_poi_target = Target(x=projected_poi[0], y=projected_poi[1], z=0, name="PPOI", type="ProjectedPOI",
+                                      index=1,
+                                      fm_focus_position=0)
+        self._tab_data_model.projected_points.append(projected_poi_target)
+        self.correlation_target.fib_projected_pois = [projected_poi_target]
+        # Display the output in the relevant views
+        self._viewports[1].canvas.update_drawing()
```

Comment:
Do NOT place a call to the GUI in the computation function. Instead, move it to the caller (ie, `_process_latest_change`). 
Also, `update_drawing()` touches the GUI. So it could crash the GUI. Instead, you can use `.Refresh()`, which is safe to call from a separate thread (and ensures that the canvas doesn't update too frequently, but here it is not an issue as the function is already rate-limited).